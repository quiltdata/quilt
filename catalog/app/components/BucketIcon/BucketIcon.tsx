import cx from 'classnames'
import * as React from 'react'
import * as M from '@material-ui/core'

import bucketIcon from './bucket.svg'

const useStyles = M.makeStyles((t) => ({
  root: {
    height: t.spacing(4),
    width: t.spacing(4),
  },
}))

interface BucketIconProps
  extends React.DetailedHTMLProps<
    React.ImgHTMLAttributes<HTMLImageElement>,
    HTMLImageElement
  > {
  alt: string
  className?: string
  classes?: {
    custom?: string
    stub?: string
  }
  src: string
}

export default function BucketIcon({
  alt,
  className: optClassName,
  classes: optClasses,
  src,
  ...props
}: BucketIconProps) {
  const classes = useStyles()

  const className = src
    ? cx(classes.root, optClasses?.custom, optClassName)
    : cx(classes.root, optClasses?.stub, optClassName)

  return <img alt={alt} className={className} src={src || bucketIcon} {...props} />
}
