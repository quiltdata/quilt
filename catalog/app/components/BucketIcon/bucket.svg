<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 149 149" style="enable-background:new 0 0 149 149" xml:space="preserve"><circle cx="74.5" cy="74.5" r="71" style="fill:#fff"/><path d="M74.5 149C33.4 149 0 115.6 0 74.5S33.4 0 74.5 0 149 33.4 149 74.5 115.6 149 74.5 149zm0-142C37.3 7 7 37.3 7 74.5S37.3 142 74.5 142 142 111.7 142 74.5 111.8 7 74.5 7z" style="fill:#d6d6d6"/><path d="m112 85-5.3-3.8 4.4-35.9c.1-1.1-.2-2.3-1-3.1-.8-.9-1.8-1.3-3-1.3H42.8c-1.1 0-2.2.5-3 1.3-.8.9-1.1 2-1 3.1l7.7 63.4c.2 2 1.9 3.5 4 3.5h48.8c2 0 3.7-1.5 4-3.5l2.2-18.4 1.8 1.3c.7.5 1.5.7 2.3.7 1.2 0 2.5-.6 3.3-1.7 1.3-1.9.9-4.4-.9-5.6zm-16.2 19.2H54.1l-6.7-55.4h55.3l-3.3 27.1-17.3-12.2v-.1c0-4.1-3.3-7.4-7.4-7.4s-7.4 3.3-7.4 7.4 3.3 7.4 7.4 7.4c1.1 0 2-.2 3-.6L98.1 85l-2.3 19.2z" style="fill:#545454"/></svg>