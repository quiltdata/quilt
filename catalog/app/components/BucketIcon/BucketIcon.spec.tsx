import * as React from 'react'
import renderer from 'react-test-renderer'

import BucketIcon from './'

describe('components/BucketIcon', () => {
  it('should render default when no src', () => {
    const tree = renderer.create(<BucketIcon alt="No src" src="" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render custom src', () => {
    const tree = renderer
      .create(<BucketIcon alt="Custom src" src="https://custom-src" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  describe('class names', () => {
    const className = 'A'
    const classes = {
      custom: 'C',
      stub: 'S',
    }
    it('should apply className', () => {
      const tree = renderer
        .create(
          <BucketIcon
            alt="Set className"
            className={className}
            src="https://custom-src"
          />,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('should apply `` className if src is set', () => {
      const tree = renderer
        .create(
          <BucketIcon
            alt="Custom className"
            className={className}
            classes={classes}
            src="https://custom-src"
          />,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it('should apply `stub` className if no src', () => {
      const tree = renderer
        .create(
          <BucketIcon
            alt="Stub className"
            className={className}
            classes={classes}
            src=""
          />,
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
