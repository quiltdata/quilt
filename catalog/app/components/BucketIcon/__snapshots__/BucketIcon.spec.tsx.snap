// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BucketIcon class names should apply \`\` className if src is set 1`] = `
<img
  alt="Custom className"
  className="makeStyles-root-1 C A"
  src="https://custom-src"
/>
`;

exports[`components/BucketIcon class names should apply \`stub\` className if no src 1`] = `
<img
  alt="Stub className"
  className="makeStyles-root-1 S A"
  src="IMAGE_MOCK"
/>
`;

exports[`components/BucketIcon class names should apply className 1`] = `
<img
  alt="Set className"
  className="makeStyles-root-1 A"
  src="https://custom-src"
/>
`;

exports[`components/BucketIcon should render custom src 1`] = `
<img
  alt="Custom src"
  className="makeStyles-root-1"
  src="https://custom-src"
/>
`;

exports[`components/BucketIcon should render default when no src 1`] = `
<img
  alt="No src"
  className="makeStyles-root-1"
  src="IMAGE_MOCK"
/>
`;
